Installing-Oppia-(Mac-OS;-Python-3).md

```diff
@@ -9,62 +9,81 @@
 
-    * Download `plistlib.py`
+    If you use Homebrew to install any Python development dependencies for pyenv (discussed below), you will need to install and use Homebrew in this terminal as well. Note that while you can have a Homebrew installation for Apple Silicon and another for Intel architectures installed simultaneously, pyenv is not smart enough to pick the Intel dependencies if both are present, so we recommend using an Intel installation of Homebrew exclusively.
 
-   ```console
-   $ curl https://raw.githubusercontent.com/python/cpython/2.7/Lib/plistlib.py -o 
-   /Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/plistlib.py
-   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
-                                 Dload  Upload   Total   Spent    Left  Speed
-   100 15185  100 15185    0     0  44274      0 --:--:-- --:--:-- --:--:-- 45193
-   ```
-    * Now try doing `python --version` on your command-line, and you should see version `2.7` printed.
+3. Follow the installation instructions below in your Rosetta terminal.
 
-2. Ensure that you have [Python 3.8](https://www.python.org/downloads/release/python-3812/) installed (Note: you can check this by running `python3 --version`). If Python 3.8 is not installed, download and run the latest Python 3.8 installer from https://www.python.org/downloads/mac-osx/.
+## Install prerequisites
+
+Oppia relies on a number of programs and third-party libraries. Many of these libraries are downloaded automatically for you when you first run the `start.py` script provided with Oppia. However, there are some things that you will need to do beforehand:
+
+1. Install [version 8+ of the Java Runtime Environment (JRE)](https://www.java.com/en/download/).
 
-3. Download [git](http://git-scm.com/download/mac), then run the package and follow instructions. This allows you to store the source in version control.
+2. Install [direnv](https://direnv.net/).
+
+   After finishing with the official installation instructions, you should also create a `~/.direnvrc` file with the following content:
+
+   ```bash
+   use_python() {
+     local python_root=$(pyenv root)/versions/$1
+     load_prefix "$python_root"
+     if [[ -x "$python_root/bin/python" ]]; then
+       layout python "$python_root/bin/python"
+     else
+       echo "Error: $python_root/bin/python can't be executed."
+       exit
+     fi
+   }
+   ```
 
-**Note:** While #1 above is only for those having MacOS >= 12.3, #2 and #3 should be done by everyone using a Mac for setting up Oppia. 
+3. Download [git](http://git-scm.com/download/mac), then run the package and follow instructions. This allows you to store the source in version control. Note that you can also install Apple's command line tools to get git instead.
```

Comment:
Can we put `brew install git` to ensure everyone is installing from the same source? It will be easier to debug if devs face an issue with git.

Reply:
I don't think I understand. Why does `brew install git` result in everyone installing from the same source, while `install from git-scm.com` doesn't? If your point is just that we should give one source instead of the note about Apple's command line tools, then why install from Homebrew over the two sources currently listed?

I actually think installing from Apple's command-line tools makes the most sense. Users will probably have to install those anyway, and there's no reason to have multiple versions of git on your system at once. wdyt?

Reply:
My bad, you're right. However, since this is for Mac OS only, I think it would be easier for the devs to simply copy the command directly from our wiki rather than going to a new website and figuring out how they want to install it.

Reply:
I'm still unclear on what approach you think is best. Here are the options:

1. Install from git-scm.com. This can't be done just by running a terminal command because the installer has a GUI. Also, we want people to get the latest version, but we would have to pick a version to give a `curl` or `wget` download command.
2. Install from Homebrew. This requires that they install Homebrew, so it's also not as simple as copying a command from the wiki. They would need to (a) run the Homebrew installation command, (b) modify the *rc file for their shell, and (c) install git.
3. Install Apple's command-line tools. This can be started from the command-line, but the installer is again graphical, so the command-line is insufficient.

Reply:
Hmmm, installing from the cmd is more tedious than I had anticipated. But thinking again, contributors should be able to install git on their own. Hence let's go with the original text you had written.